File: android/app/src/main/kotlin/uk/co/twinscrollgridbalancer/tsgbheater/data/store/BoundDevice.kt
package uk.co.twinscrollgridbalancer.tsgbheater.data.store

import kotlinx.serialization.Serializable
import uk.co.twinscrollgridbalancer.tsgbheater.protocol.ProtocolKind

// One persisted pairing. The [protocol] field drives which driver
// BleManager instantiates when this device is opened — see
// ProtocolRegistry. Existing JSON entries from before this field
// existed deserialize with the default (HEATGENIE), which is correct
// since every pre-multi-protocol pairing was a HeatGenie heater.
@Serializable
data class BoundDevice(
    val mac: String,
    val name: String,
    val lastSeenAtMs: Long,
    val autoStartStopEnabled: Boolean = false,
    val protocol: ProtocolKind = ProtocolKind.HEATGENIE,
    // --- Fuel tank config (per-heater) ---
    // Mirrors the C# BoundDevice on the Windows side. The defaults
    // match the user's current setup: 5 L tank, 0.15–0.55 L/h
    // consumption across 10 linearly-interpolated gears.
    val tankLitres: Double = 5.0,
    val consumptionLowLph: Double = 0.15,
    val consumptionHighLph: Double = 0.55,
)

File: windows/src/Ember/Data/BoundDevice.cs
using Ember.Protocol;

namespace Ember.Data;

// Mirror of android/.../data/store/BoundDevice.kt — same shape so a future
// sync feature can move records between phones and the laptop trivially.
//
// The Protocol field drives which driver HeaterClient instantiates when
// this device is opened. Existing JSON entries from before this field
// existed deserialize with the HeatGenie default (System.Text.Json fills
// missing record params with the parameter default), which is correct
// since every pre-multi-protocol pairing was a HeatGenie heater.
public sealed record BoundDevice(
    string       Mac,
    string       Name,
    long         LastSeenAtMs,
    bool         AutoStartStopEnabled = false,
    ProtocolKind Protocol             = ProtocolKind.HeatGenie,
    // --- Fuel tank config (per-heater) ---
    // Total tank capacity, in litres. The default of 5L matches the
    // common 5-litre canister the user runs today; each heater can have
    // a different tank.
    double       TankLitres           = 5.0,
    // Consumption range from the manual: at lowest gear (1) the
    // burner uses ConsumptionLowLph litres/hour; at highest gear (10),
    // ConsumptionHighLph. The FuelTracker linearly interpolates between
    // these for whichever gear is reported in telemetry.
    double       ConsumptionLowLph    = 0.15,
    double       ConsumptionHighLph   = 0.55);
